Documentation/en-us/TestUsingTestDoubles.md

```diff
@@ -0,0 +1,131 @@
+
+* Display data from internet in ArticleViewController
+* Custom class implements DataProviderProtocol is responsible for getting data.
+
+DataProviderProtocol is defined as follows,
+
+```swift
+// Swift
+protocol DataProviderProtocol: class {
+    func fetch(callback: (data: String) -> Void)
+}
+```
+`fetch()` gets data from internet and output data with callback block.
+
+Here's DataProvider which implements DataProviderProtocol.
+
+```swift
+// Swift
+class DataProvider: NSObject, DataProviderProtocol {
+    func fetch(callback: (data: String) -> Void) {
+        let url  = NSURL(string: "http://example.com/")!
+        let session = NSURLSession(configuration: NSURLSessionConfiguration.defaultSessionConfiguration())
+        let task    = session.dataTaskWithURL(url, completionHandler: {
+            (data, resp, err) in
+            let string = NSString(data:data!, encoding:NSUTF8StringEncoding) as! String
+            callback(data: string)
+        })
+        task.resume()
+    }
+}
+```
+In our scenario, `fetch()` is called in `viewDidLoad()` of ViewController.
+
+Code is below.
+
+```swift
+// Swift
+class ViewController: UIViewController {
+    @IBOutlet weak var resultLabel: UILabel!
+    private var dataProvider: DataProviderProtocol?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        dataProvider = dataProvider ?? DataProvider()
+
+        dataProvider?.fetch({ [unowned self] (data) -> Void in
+            self.resultLabel.text = data
+        })
+    }
+}
+```
+
+#### Writing test with Mock of DataProviderProtocol
+
+ArticleViewController depends on DataProviderProtocol.
+Here, create Mock which inherits DataProviderProtocol in Test Targets in order to test ViewController.
+
+```swift
+// Swift
+class MockDataProvider: NSObject, DataProviderProtocol {
+    var fetchCalled = false
+    func fetch(callback: (data: String) -> Void) {
+        fetchCalled = true
+        callback(data: "foobar")
+    }
+}
+```
+
+`fetchCalled` property is deinfed in Mock. `fetchCalled` is set to `true` when `fetch()` called.
+Ready to run test!
+
+This test verifies that `When ViewController is loaded, ViewController calls dataProvider.fetch()`.
+
+```swift
+// Swift
+override func spec() {
+    describe("view controller") {
+        it("fetch data with data provider") {
+            let mockProvier = MockDataProvider()
+            let viewController = UIStoryboard(name: "Main", bundle: nil).instantiateViewControllerWithIdentifier("ViewController") as! ViewController
+            viewController.dataProvier = mockProvier
+            
+            expect(mockProvier.fetchCalled).to(equal(false))
+
+            let _ = viewController.view
+            
+            expect(mockProvier.fetchCalled).to(equal(true))
+        }
+    }
+}
+```
+
+Mock makes us easy to verify behavior which works with other objects.
+
+For more details about writing test, see https://realm.io/news/testing-in-swift/ .
```

Comment:
I am planning on reviewing this additional documentation in #487, but I don't think you meant to include it in this pull request. Could you take it out?